### PR TITLE
Fix for memberships on group transfer job

### PR DESCRIPTION
**Pull request by ksierks**
## What does this PR do and why?
The job fixes the memberships after a group transfer, so that the access level is ascending through its descendants.
Project transfers is out of scope in this PR.
Fixes part of https://github.com/phac-nml/irida-next/issues/226.

## Screenshots or screen recordings
N/A

## How to set up and validate locally
1. Verify the UpdateMembershipsJob is called when a group transfer is initiated. 
  a. Please see instructions from #264
2. Call the UpdateMembershipsJob from console.
  a. Open a rails console. Run `rails c` in a terminal.
  b. Call `UpdateMembershipsJob.perform_now` to start a job. Pass it a list of membership ids.
  c. Verify the memberships were updated.

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.

- [x] I have evaluated the [PR acceptance checklist](https://phac-nml.github.io/irida-next/docs/development/development_processes/code_review#acceptance-checklist) for this PR.

### CODE REVIEW

**Comment by deepsidhu85** [comment]
After looking through the changes in the PR I think there's a few things we can change to clean up some things like having to skip/set the call back in `member.rb`.

Here is my thoughts:

Since we created a job to update the descendant memberships on group/project transfer, we should have it so that the members update service also just queues a job to update the memberships.
So in `members/update_service.rb` within the execute method we want to get the result of the `member.update` and then queue up a job to update the memberships if the `member.update` was successful. Then we want to return the result of `member.update`
With this in `member.rb` we can remove the callback `after_update` for the descendant memberships

In `test/jobs/update_memberships_job_test.rb` we can remove the skipping of the callback in the setup and remove the teardown method completely.

Instead of setting the access level and then doing a save we should update the calls to use the Members::UpdateService for example

`valid_params = { user: @group_member.user, access_level: Member::AccessLevel::MAINTAINER }`
`Members::UpdateService.new(@group_member, @group_member.namespace, @user, valid_params).execute`

For the tests we can wrap the internals of the test all within a `perform_enqueued_jobs do....end` block. Calling the Members::UpdateService can be put on the lines right above where we expect the values to have changed


With the changes above we will also need to update `test/services/members/update_service_test.rb` so when the Members::UpdateService is called we need to put a `perform_enqueued_jobs` right below it

Also, we can just include the ActiveJob::TestHelper in `test/test_helper.rb`

File: test/services/groups/transfer_service_test.rb [hunk 1]
```diff
@@ -4,82 +4,39 @@
 
 module Groups
   class TransferServiceTest < ActiveSupport::TestCase
+    include ActiveJob::TestHelper
```
Comment (deepsidhu85) [inline test/services/groups/transfer_service_test.rb:7]: As per the general comment on the PR this can be moved into `test/test_helper.rb`

**Review by deepsidhu85 — CHANGES_REQUESTED** [review]
This is working as expected but I provided some feedback in a general comment on the PR for things we can change/cleanup

File: test/services/groups/transfer_service_test.rb (hunk 1 above)
Comment (ksierks) [reply test/services/groups/transfer_service_test.rb]: Please see 0ef0afd6a0e156a6483e4434ea25dde01d85eebd.

File: app/services/groups/transfer_service.rb [hunk 2]
```diff
@@ -19,12 +19,12 @@ def execute(new_namespace) # rubocop:disable Metrics/AbcSize
         raise TransferError, I18n.t('services.groups.transfer.namespace_group_exists')
       end
 
-      group_ancestor_members_user_ids = Member.for_namespace_and_ancestors(@group).select(:user_id)
-      update_members = Member.for_namespace_and_ancestors(new_namespace).where(user_id: group_ancestor_members_user_ids)
+      group_ancestor_member_user_ids = Member.for_namespace_and_ancestors(@group).select(:user_id)
+      new_namespace_member_ids = Member.for_namespace_and_ancestors(new_namespace).where(user_id: group_ancestor_member_user_ids).select(&:id) # rubocop:disable Layout/LineLength
```
[the remark this answers is not in the document]
Comment (ksierks) [reply app/services/groups/transfer_service.rb]: Please see 1349935d2b0305ac3128821f7ce5200f27df0bb4.

**Comment by ksierks** [comment]
> @ksierks After looking through the changes in the PR I think there's a few things we can change to clean up some things like having to skip/set the call back in `member.rb`.
> 
> Here is my thoughts:
> 
> Since we created a job to update the descendant memberships on group/project transfer, we should have it so that the members update service also just queues a job to update the memberships. So in `members/update_service.rb` within the execute method we want to get the result of the `member.update` and then queue up a job to update the memberships if the `member.update` was successful. Then we want to return the result of `member.update` With this in `member.rb` we can remove the callback `after_update` for the descendant memberships
> 
> In `test/jobs/update_memberships_job_test.rb` we can remove the skipping of the callback in the setup and remove the teardown method completely.
> 
> Instead of setting the access level and then doing a save we should update the calls to use the Members::UpdateService for example
> 
> `valid_params = { user: @group_member.user, access_level: Member::AccessLevel::MAINTAINER }` `Members::UpdateService.new(@group_member, @group_member.namespace, @user, valid_params).execute`
> 
> For the tests we can wrap the internals of each test within a `perform_enqueued_jobs do....end` block. Calling the Members::UpdateService can be put on the lines right above where we expect the values to have changed
> 
> With the changes above we will also need to update `test/services/members/update_service_test.rb` so when the Members::UpdateService is called we need to put a `perform_enqueued_jobs` right below it
> 
> Also, we can just include the ActiveJob::TestHelper in `test/test_helper.rb`

Hopefully, I addressed everything in e500611feb682f07b21d52e0b35a28adec66dc25, except "Instead of setting the access level and then doing a save we should update the calls to use the Members::UpdateService". I am only wanting to test the job in update_memberships_job_test.rb. If I call Members::UpdateService.execute, it calls the job that I'm trying to test. It's a little too cyclical.

**Review by deepsidhu85 — APPROVED** [review]

**Review by ChrisHuynh333 — APPROVED** [review]
Works as described, nothing to add on my end!

**Pull request merged**